File: server/Models/Comment.cs
﻿using System;

namespace Models
{
    public class Comment
    {
        public int CommentId { get; set; }
        public string Content { get; set; }
        public DateTime CommentDate { get; set; }

        // Navigation Properties
        public string StudentId { get; set; }
        public Student Student { get; set; }
        public int VideoId { get; set; }  // Foreign Key
        public Video Video { get; set; }
        public string RecruiterId { get; set; }  // Foreign Key
        public Recruiter Recruiter { get; set; }
    }
}

File: client/src/models/Comment.ts
import { Video } from './Video';
import Recruiter  from './Recruiter';
import  Student  from './Student';

export default interface Comment {
    commentId: number;
    content: string;
    commentDate: string;
    studentId: string; // Reference to the student whose video is commented on
    videoId: number; // Reference to the video being commented on
    recruiterId: string; // Reference to the recruiter making the comment
    video?: Video;
    recruiter?: Recruiter;
    student?: Student;
}
